tensorrt_llm/_torch/pyexecutor/sampler.py

```diff
@@ -583,14 +590,15 @@ def beam_width(scheduled_requests: ScheduledRequests) -> int:
             return req.sampling_config.beam_width
         raise ValueError("No beam width found")
 
+    @override
     def sample_async(self, scheduled_requests: ScheduledRequests,
                      model_outputs) -> SampleStateTRTLLM:
         batch_size = scheduled_requests.batch_size
         beam_width = self.beam_width(scheduled_requests)
 
         logits = model_outputs["logits"].reshape((batch_size, beam_width, -1))
 
-        self.setup_sampler_step(scheduled_requests.context_requests)
+        self.setup_sampler_step(scheduled_requests)
```

Comment:
Please note that this change changes behaviour. In the previous incarnation, only the context requests were passed. In the latter, both context and gen requests are passed.